include/fiction/algorithms/simulation/sidb/quickexact.hpp

```diff
@@ -0,0 +1,506 @@
+//
+// Created by Jan Drewniok on 18.12.22.
+//
+
+#ifndef FICTION_QUICKEXACT_HPP
+#define FICTION_QUICKEXACT_HPP
+
+#include "fiction/algorithms/simulation/sidb/energy_distribution.hpp"
+#include "fiction/algorithms/simulation/sidb/enum_class_exhaustive_algorithm.hpp"
+#include "fiction/algorithms/simulation/sidb/minimum_energy.hpp"
+#include "fiction/algorithms/simulation/sidb/sidb_simulation_parameters.hpp"
+#include "fiction/algorithms/simulation/sidb/sidb_simulation_result.hpp"
+#include "fiction/technology/charge_distribution_surface.hpp"
+
+#include <fmt/format.h>
+#include <mockturtle/utils/stopwatch.hpp>
+
+#include <algorithm>
+#include <cstdint>
+#include <unordered_map>
+#include <vector>
+
+namespace fiction
+{
+
+/**
+ * An enumeration of modes to use for the QuickExact algorithm.
+ */
+enum class automatic_base_number_detection
+{
+    /**
+     * It automatically detects if a 2-state simulation is sufficient or if a 3-state simulation is required. It checks
+     * if a positive charge could occur due to maximum band bending. If this mode is active, 3-state simulation could be
+     * run even if base_num = 2 was set in the physical parameters.
+     */
+    ON,
+    /**
+     * The base number from the physical parameter is used for the simulation.
+     */
+    OFF
+};
+/**
+ * This struct stores the parameters for the `quickexact` algorithm.
```

Comment:
I've noticed that you've been a bit inconsistent with the *QuickExact* naming. I've found the following spellings: *QuickExact*, QuickExact, Quickexact, quickexact, `quickexact`. I've tried to streamline, but still not sure, which one to prefer. I would say *QuickExact*.